packages/labextension/pyproject.toml

```diff
@@ -0,0 +1,64 @@
+[build-system]
+requires = ["hatchling>=1.5.0", "jupyterlab>=4.0.0,<5", "hatch-nodejs-version"]
+build-backend = "hatchling.build"
+
+[project]
+name = "jupyterlab_jupytext"
+readme = "README.md"
+license = { file = "LICENSE" }
+requires-python = ">=3.8"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: MIT License",
+    "Environment :: Console",
+    "Framework :: Jupyter",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Topic :: Text Processing :: Markup",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dependencies = [
+    "jupyterlab>=4.0.0,<5",
+]
+dynamic = ["version", "description", "authors", "urls", "keywords"]
```

Comment:
Can the `version` in nodejs be taken dynamically? Preferably you take them both from the top-level git repo.

Also add a `.git_archival.txt` for dynamic git versioning.

Reply:
Currently, core `jupytext` and `jupyterlab-jupytext` are not on the same version. For the frontend extension, python package will take the version specified in `package.json` and so, yeah, the version is set dynamically from npm package. The frontend extension is a standalone package by itself and it should be able to build without having any dependencies on the top level repo.

Reply:
I think the versions of `jupytext` and `jupyterlab-jupytext` could remain different. Would it make sense to give a version number `4.0.0` to this new `jupyterlab-jupytext` package to indicate that it works with Jupyter Lab 4? (Also, how complicated would it be to provide a version 3.0 of `jupyterlab-jupytext` for the users that have not yet switched to JLab 4?)

Reply:
> If I am right, currently the extension's version is on 1.x. So, I dont know if it is a good idea to skip versions and jump to 4.x directly.

This is fine because the ownership of the package has not changed (there was a fiasco with `build` package upgrading to `1.0` on PyPI). I think it is good to have it mirror at least the major versions, unless they can be completely decoupled (not the case right now because of the splitting).

> In any case, if we are going to provide jupytext_config as a module, we are sort of modifying the API (even though it is very minor).

It is additive so a minor bump is sufficient for this change. There would be a major bump due to the re-arranging of the project though.

> Yes, we can publish a jupyterlab-jupytext compatible with JupyterLab 3.x with this PR. And in a separate PR, we bump the versions of npm dependencies of extension to JupyterLab 4.x and make a new release.

Yes, we should make a separate PR for the current release cycle on both minor and patch versions to announce the future organization, and upper-pinning the major versions.

Reply:
Re version numbers, I think I would like to label the first release of the `jupyterlab-jupytext` extension that will work exclusively with Jupyter Lab 4 as `4.0.0`. Until now I've been using `1.3.x` for Jupyter Lab 3 but that's not very standard I guess.

Re `jupytext==1.16` or `jupytext==2.0` I was thinking of going with `jupytext==1.16` for now, and keep `jupytext==2.0` for the day when we can provide a text format with outputs (but I am not sure when or if this will actually happen :smiley: )

Reply:
Anyways we can defer this discussion to a new PR to bump the versions of both `jupytext` and `jupyterlab-jupytext` npm extension.